### ISSUE TITLE
Titles and parent IDs or types don't play nicely together

### ISSUE THREAD

**Issue by Bjwebb** [OpenDataServices/flatten-tool#56]
This is due to https://github.com/open-contracting/flattening-ocds/blob/master/flattening_ocds/input.py#L31 which replaces all colons with slashes.

A temporary fix to allow users to supply a non-title header for parent-ids or typed columns would be to not perform the conversion if we find a slash.

A longer term solution might be to use a different separator for this purpose when titles are used.

**Comment by practicalparticipation** [comment]
If possible I think it would be useful to re-examine the way we're handling the sub-sheet to parent sheet referencing... as not sure we've hit the right spot yet in terms of maximum understandability to the basic user. 

Worth a quick discussion when we next meet?

**Comment by Bjwebb** [comment]
Yes, definitely worth discussing,

**Comment by Bjwebb** [comment]
We discussed this briefly, deciding to stick with the current approach.

I'm fairly sure that there are some implications in terms of Cove, and ability to use type annotations in the headers. Current task is for me to write this up.

**Comment by Bjwebb** [comment]
With https://github.com/OpenDataServices/flatten-tool/pull/89 we've removed types annotations from the main sheet. We hope to extend this to all other sheets https://github.com/OpenDataServices/flatten-tool/issues/90 -> this will also have implications for how sub-sheet to main-sheet referencing works; the task in that issue to write up what the possible approaches might be, and what the impacts of those are.

Closing this issue as I think it's been superseded.
